src/ResourceManager/Network/Commands.Network/NetworkWatcher/SetAzureNetworkWatcherConfigFlowLogCommand.cs

```diff
@@ -125,6 +245,50 @@ public override void Execute()
                         parameters.RetentionPolicy.Days = this.RetentionInDays;
                     }
 
+                    if (ParameterSetName.Contains(WithTA))
+                    {
+                        parameters.FlowAnalyticsConfiguration = new MNM.TrafficAnalyticsProperties();
+                        parameters.FlowAnalyticsConfiguration.NetworkWatcherFlowAnalyticsConfiguration = new MNM.TrafficAnalyticsConfigurationProperties();
+
+                        parameters.FlowAnalyticsConfiguration.NetworkWatcherFlowAnalyticsConfiguration.Enabled = this.EnableTrafficAnalytics.IsPresent;
+
+                        if (ParameterSetName.Contains(TAByDetails))
+                        {
+                            if (this.WorkspaceResourceId == null || this.WorkspaceGUId == null || this.WorkspaceLocation == null)
+                            {
+                                throw new System.ArgumentException("Either the Workspace parameter or all of WorkspaceResourceId,WorkspaceGUId,WorkspaceLocation must be provided");
+                            }
+
+                            string[] workspaceDetailsComponents = this.WorkspaceResourceId.Split('/');
```

Comment:
You can use this: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Utilities/Models/ResourceIdentifier.cs

Reply:
None of these APIs validate check of exact 8 components (They check >= 8). Is it advisabel to use that check still?